etrago/tools/extendable.py

```diff
@@ -66,56 +66,93 @@ def extendable(self, line_max):
         Overall container of PyPSA
     """
     network = self.network
+    extendable_settings = self.args["extendable"]
+
     
-    if not 'as_in_db' in self.args['extendable']:
+    if not 'as_in_db' in extendable_settings["extendable_components"]:
         network.lines.s_nom_extendable = False
         network.transformers.s_nom_extendable = False
         network.links.p_nom_extendable = False
         network.storage_units.p_nom_extendable = False
         network.stores.e_nom_extendable = False
         network.generators.p_nom_extendable = False
-
-
-    if 'network' in self.args['extendable']:
+        
+    if not grid_max_abs_D==None:
+        buses = network.buses[network.buses.country=='DE']
+        
+        line_max_abs(network=network, buses=buses,
+                     line_max_abs=grid_max_abs_D)
+        
+        transformer_max_abs(network=network, buses=buses)
+
+        network.links.loc[(network.links.bus0.isin(buses.index)) &
+                (network.links.bus1.isin(buses.index)),
+                  'p_nom_max'] = grid_max_abs_D['dc']
+        
+    if not grid_max_abs_foreign==None:
+        foreign_buses = network.buses[network.buses.country!='DE']
+        line_max_abs(network=network, buses=foreign_buses,
+                     line_max_abs=grid_max_abs_foreign)
+        
+        transformer_max_abs(network=network, buses=foreign_buses)
+               
+        network.links.loc[(network.links.bus0.isin(foreign_buses.index)) |
+                              (network.links.bus1.isin(foreign_buses.index)),
+                          'p_nom_max'] = grid_max_abs_foreign['dc']  
+
+        
+    if not grid_max_D == None:
+        buses = network.buses[network.buses.country=='DE']
+        
+        network.lines.loc[(network.lines.bus0.isin(buses.index)) &
+                    (network.lines.bus1.isin(buses.index)),
+                    's_nom_max'] = grid_max_D * network.lines.s_nom  
+        
+        network.transformers.loc[network.transformers.bus0.isin(
+            buses.index),'s_nom_max'] = grid_max_D * network.transformers.s_nom
+        
+        network.links.loc[(network.links.bus0.isin(buses.index)) &
+        (network.links.bus1.isin(buses.index)),
+          'p_nom_max'] = grid_max_D * network.links.p_nom
```

Comment:
This sets upper limits for all links in Germany, right? To exclude the sector-coupling links, the` buses` in line 105 could get an additional filter `carrier == 'AC'`

Reply:
Good point! Forgot about the other sectors :flushed: I will adjust that!

Reply:
solved by d1bcb57b1b2bd48f95adf08ce183acae606315ba